### ISSUE TITLE
How to support packages that do not conform to our "standard layout" (to be specified...)

### ISSUE THREAD

**Issue by certik** [fortran-lang/fpm#6]
How to support packages that do not conform to our "standard layout" (to be specified...). Some examples of such a package would be reference Lapack, or Arpack. The way to do that is that we create a new repository, say `certik/lapack.fpm`, which will have `fpm.toml`, in there it would specify the url to the actual sources (https://github.com/Reference-LAPACK/lapack) and a build script, which would build the sources (using CMake in this case) and install them into some $PREFIX provided by `fpm` and `fpm` takes it from there. This approach also works for non Fortran packages --- the build script either builds it, or requires it from the system (where it can be provided by, e.g., Spack). Either way this is a clean way to hook this up into the `fpm` ecosystem.

**Comment by milancurcic** [comment]
Yes, I think fpm.toml for each project should include the exact commands that need to be executed to build the package, very much like we do in yml files for CI. 

That said, until a need arises, I'd suggest that we take this approach for all packages, including those built with CMake, autotools, or plain Makefiles.

If we can make a package build in CI, we should be able to build it by fpm by making it execute commands specified in fpm.toml.

> The way to do that is that we create a new repository, say certik/lapack.fpm

Does this even need to be a separate repo? Are you thinking about a repo that would be the registry of all fpm-supported packages, like what I described in #4 ?
